src/components/TodoItem.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDelete?: (todoId: number) => void;
+  todoIds: number[];
+  onClick: (todo: Todo) => void;
+  updateTitle?: (todo: Todo, title: string) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  todoIds,
+  onClick,
+  updateTitle,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleField.current && isEditing) {
+      titleField.current.focus();
+    }
+  }, [isEditing]);
+
+  const deleteTodo = () => {
+    if (onDelete) {
+      onDelete(todo.id);
+    }
+  };
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleOnKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      setIsEditing(false);
```

Comment:
Looks like here need to make also setNewTitle(todo.title), because exists some bug.
Steps to reproduce:
1. Start edit some todo
2. Press escape
3. Make double click for start edit again this todo. Expected to show actual todo.title. Actual shows todo.title which was before press escape.